merlin/systems/dag/dictarray.py

```diff
@@ -13,44 +13,149 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+from enum import Enum
 from typing import Dict, Optional
 
 import numpy as np
 
-from merlin.core.protocols import SeriesLike, Transformable
+from merlin.core.dispatch import HAS_GPU
+from merlin.core.protocols import SeriesLike
+
+if HAS_GPU:
+    import cupy
+else:
+    cupy = None
+
+
+class Device(Enum):
+    CPU = 0
+    GPU = 1
 
 
 class Column(SeriesLike):
     """
-    A simple wrapper around an array of values
+    A simple wrapper around an array of values. This has an API that's just similar enough to
+    Pandas and cuDF series to be relatively interchangeable from the perspective of the Merlin DAG,
+    but no more. As more methods get added to this class, it gets closer and closer to actually
+    *being* a Pandas/cuDF Series (at which point there's no advantage to using this.) So: keep
+    this class as small as possible.
     """
 
-    def __init__(self, values):
+    def __init__(self, values, row_lengths=None):
```

Comment:
We chose to go with row lengths here, but it could equally well be offsets. It's slightly easier to sum the row lengths to get the offsets than to do the subtractions to get the lengths from the offsets, so we did the easy thing here.